core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/GoogleProjectRegistrationApiService.scala

```diff
@@ -20,22 +25,45 @@ trait GoogleProjectRegistrationApiService extends UserInfoDirectives {
   def googleProjectRegistrationRoutes(otelContext: Context = Context.root()): server.Route =
     requireUserInfo(Option(otelContext)) { userInfo =>
       val ctx = RawlsRequestContext(userInfo, Option(otelContext))
-      path("googleProjects") {
-        put {
-          entity(as[GoogleProjectRegistration]) { entity =>
-            complete {
-              googleProjectRegServiceConstructor(ctx)
-                .registerGoogleProject(
-                  entity
-                )
-                .map {
-                  case None          => StatusCodes.OK -> None
-                  case Some(project) => StatusCodes.Created -> Some(project)
+      pathPrefix("googleProjects") {
+        pathEnd {
+          put {
+            entity(as[GoogleProjectRegistration]) { entity =>
+              complete {
+                googleProjectRegServiceConstructor(ctx)
+                  .registerGoogleProject(
+                    entity
+                  )
+                  .map {
+                    case None          => StatusCodes.OK -> None
+                    case Some(project) => StatusCodes.Created -> Some(project)
+                  }
+              }
+            }
+          } ~
+            get {
+              parameters("billingProjectId".optional) { billingProjectId =>
+                onSuccess(
+                  googleProjectRegServiceConstructor(ctx)
+                    .getGoogleProjects(billingProjectId.map(RawlsBillingProjectName))
+                ) { projects =>
+                  if (projects.isEmpty) complete(StatusCodes.NotFound -> None)
+                  else complete(StatusCodes.OK -> projects)
                 }
+              }
+            }
+        } ~
+          path(Segment) { googleProjectId =>
+            get {
+              onSuccess(
+                googleProjectRegServiceConstructor(ctx)
+                  .getGoogleProjectById(GoogleProjectId(googleProjectId))
+              ) {
+                case Some(projectId) => complete(StatusCodes.OK -> projectId)
+                case None            => complete(StatusCodes.NotFound)
```

Comment:
Both these APIs should use the Rawls-standard "does not exist or you do not have permission to use it" message. To make this happen you probably will need to throw an error from `GoogleProjectRegistrationService` instead of handling the Some/None cases here.